omc3/tune_analysis/kick_file_modifiers.py

```diff
@@ -0,0 +1,258 @@
+    return kickac_df
+
+
+def add_odr(kickac_df, odr_fit, action_plane, tune_plane, corrected=False):
+    """ Adds the odr fit of the (un)corrected data to the header of the kickac.
+
+    Args:
+        kickac_df: Dataframe containing the data
+        odr_fit: odr-fit data (definitions see ``detuning_tools.py``)
+        action_plane: Plane of the action
+        tune_plane: Plane of the tune
+        corrected: (BBQ) corrected data or uncorrected fit?
+
+    Returns:
+        Modified kick_ac
+    """
+    header_val, header_err = _get_odr_headers(corrected)
+    for idx in range(len(odr_fit.beta)):
+        kickac_df.headers[header_val(action_plane, tune_plane, idx)] = odr_fit.beta[idx]
+        kickac_df.headers[header_err(action_plane, tune_plane, idx)] = odr_fit.sd_beta[idx]
+    return kickac_df
+
+
+def add_total_natq_std(kickac_df):
+    """ Add the total standard deviation of the natural tune to the kickac.
+    The total standard deviation is here defined as the standard deviation of the measurement
+    plus the standard deviation of the moving average.
+
+    Args:
+        kickac_df: Dataframe containing the data
+
+    Returns:
+        Modified kick_ac
+    """
+    for plane in PLANES:
+        kickac_df[COL_NATQ_CORRSTD(plane)] = np.sqrt(
+            np.power(kickac_df[COL_NATQ_STD(plane)], 2) +
+            np.power(kickac_df[COL_MAV_STD(plane)], 2)
+        )
+    return kickac_df
+
+
+# Data Extraction ##############################################################
+
+
+def get_odr_data(kickac_df, action_plane, tune_plane, order, corrected=False):
+    """ Extract the data from kickac.
+
+    Args:
+        kickac_df: Dataframe containing the data
+        action_plane: Plane of the action
+        tune_plane: Plane of the tune
+        order: Order of the odr fit
+        corrected: (BBQ) corrected data or uncorrected fit?
+
+    Returns:
+        Dictionary containing
+
+    """
+
+    header_val, header_err = _get_odr_headers(corrected)
+    odr_data = DotDict(beta=[0] * (order+1), sd_beta=[0] * (order+1))
+    for idx in range(order+1):
+        odr_data.beta[idx] = kickac_df.headers[header_val(action_plane, tune_plane, idx)]
+        odr_data.sd_beta[idx] = kickac_df.headers[header_err(action_plane, tune_plane, idx)]
+    return odr_data
+
+
+def get_ampdet_data(kickac_df, action_plane, tune_plane, corrected=False):
+    """ Extract the data needed for the (un)corrected amplitude detuning
+    from the kickac dataframe.
+
+    Args:
+        kickac_df: Dataframe containing the data
+        action_plane: Plane of the action
+        tune_plane: Plane of the tune
+
+
+    Returns:
+        Dictionary containing x,y, x_err and y_err
+
+    """
+    col_natq, col_natq_std = _get_ampdet_columns(corrected)
+
+    columns = {"x": COL_ACTION(action_plane),
+               "xerr": COL_ACTION_ERR(action_plane),
+               "y": col_natq(tune_plane),
+               "yerr": col_natq_std(tune_plane),
+               }
```

Comment:
as action plane/tune_plane is also element of (x,y), this could be a bit confusing to read, maybe just naming action and natQ instead?